pkg/apis/core/validation/validation.go

```diff
@@ -2769,14 +2777,22 @@ func gatherPodResourceClaimNames(claims []core.PodResourceClaim) sets.String {
 	return podClaimNames
 }
 
-func validatePodResourceClaim(claim core.PodResourceClaim, podClaimNames *sets.String, fldPath *field.Path) field.ErrorList {
+func validatePodResourceClaim(podMeta *metav1.ObjectMeta, claim core.PodResourceClaim, podClaimNames *sets.String, fldPath *field.Path) field.ErrorList {
 	var allErrs field.ErrorList
 	if claim.Name == "" {
 		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
 	} else if podClaimNames.Has(claim.Name) {
 		allErrs = append(allErrs, field.Duplicate(fldPath.Child("name"), claim.Name))
 	} else {
-		allErrs = append(allErrs, ValidateDNS1123Label(claim.Name, fldPath.Child("name"))...)
+		nameErrs := ValidateDNS1123Label(claim.Name, fldPath.Child("name"))
+		if len(nameErrs) > 0 {
+			allErrs = append(allErrs, nameErrs...)
+		} else if podMeta != nil && claim.Source.ResourceClaimTemplateName != nil {
+			claimName := podMeta.Name + "-" + claim.Name
+			for _, detail := range ValidateResourceClaimName(claimName, false) {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("name"), claimName, "final ResourceClaim name: "+detail))
+			}
+		}
```

Comment:
This seems to be the soul of the change.
i.e. add validation of the names "generated" for ResourceClaims from ResourceClaimTemplates embedded in a pod spec.

The mechanics of this change itself seem mostly fine. One question -- do we want to check `ValidateResourceClaimName()` also for claimNames *not* coming from a template? Or is the `ValidateDNS1123Label()` already a stricter check, so unnecessary except when we concatenate?

Also, this doesn't address the issue of the name being a simple concatenation of the podName and the claimTemplateName. But that is a bigger discussion.

Reply:
ValidateDNS1123Label is stricter. For example, it doesn't allow dots.

Reply:
Thanks for confirming.